SPR-16098: Fix Reactive JsonView + HttpEntity handling

sdeleuze opened the pull request:
This commit adds an `AbstractMessageWriterResultHandler#writeBody` variant which allows to pass the actual `MethodParameter` in order to perform proper annotation-based hint resolution with nested generics like with `ResponseEntityResultHandler`.

rstoyanchev commented on spring-webflux/src/main/java/org/springframework/web/reactive/result/method/annotation/AbstractMessageWriterResultHandler.java:
```diff
@@ -86,9 +87,33 @@ protected AbstractMessageWriterResultHandler(List<HttpMessageWriter<?>> messageW
+	 * @since 5.0.2
+	 */
+	@SuppressWarnings("unchecked")
+	protected Mono<Void> writeBody(@Nullable Object body, MethodParameter bodyParameter, ServerWebExchange exchange, @Nullable MethodParameter actualParameter) {
```
Let's put ServerWebExchange as the last argument, i.e. `(body, bodyParameter, actualParameter, exchange)`.

sdeleuze replied:
Indeed we have the same issue with `HttpEntity` I will fix that as well with a similar strategy. Notice that `@RequestBody HttpEntity<Foo> foo entity` does not seems to be currently supported, only `HttpEntity<Foo> foo entity`. Likely an `ArgumentResolver` ordering issue, I will create a JIRA issue for that.

rstoyanchev reviewed (commented):
Looks fine. Do we have the same issue with @RequestBody and HttpEntity?

sdeleuze committed:
Fix JsonView + HttpEntity Reactive handling

This commit adds
AbstractMessageReaderArgumentResolver#readBody and
AbstractMessageWriterResultHandler#writeBody variants
which allow to pass the actual MethodParameter in order
to perform proper annotation-based hint resolution with
nested generics, for example with HttpEntity.

Issue: SPR-16098

sdeleuze commented:
Merged via https://github.com/spring-projects/spring-framework/commit/c530745015d5c7031cfcb6f633f3f459ed97935d.
